src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -114,20 +122,24 @@ public async Task<IPackageSearchMetadata> GetPackageMetadataAsync(PackageIdentit
             bool includeUnlisted,
             CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
             var tasks = _sourceRepositories
                 .Select(r => GetMetadataTaskSafeAsync(()=> r.GetPackageMetadataListAsync(packageId, includePrerelease, includeUnlisted, cancellationToken)))
                 .ToArray();
 
             var completed = (await Task.WhenAll(tasks))
                 .Where(m => m != null);
 
+            cancellationToken.ThrowIfCancellationRequested();
             var packages = completed.SelectMany(p => p);
 
+            cancellationToken.ThrowIfCancellationRequested();
             var uniquePackages = packages
                 .GroupBy(
                     m => m.Identity.Version,
                     (v, ms) => ms.First());
 
+            cancellationToken.ThrowIfCancellationRequested();
             return uniquePackages;
```

Comment:
Don't cancel when the work has already been completed. See recommendation 3 in [this blog post](https://devblogs.microsoft.com/premier-developer/recommended-patterns-for-cancellationtoken/).

Reply:
Been referencing that blog, guess I got overzealous with copy/paste on this one